tests/rptest/clients/offline_log_viewer.py

```diff
@@ -42,3 +44,24 @@ def read_controller(self, node):
 
     def read_consumer_offsets(self, node):
         return self._json_cmd(node, "--type consumer_offsets")
+
+    def read_bin_topic_manifest(self, bin_data, return_legacy_format=True):
+        """
+        copy bin_data to a local file, invoke the tool with it and 
+        return a dict with the decoded result. 
+        return_legacy_format=True produces an output like the original 
+        topic_manifest.json
+        """
+        path = f"/tmp/{str(uuid.uuid4())}.topic_manifest.bin"
```

Comment:
It's surprising that we leave behind files for runs of this method. Maybe worth copying some of clients/rp_storage_tool.py regarding temporary files?

Reply:
ah, I see. I missed the `with tempfile.NamedTemporaryFile(mode='wb') as tmp`